src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -87,10 +92,16 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String replacement = "<img src=\"$1\" />";
     String textWithImagesReplaced = text.replaceAll(regex, replacement);
 
+    MutableDataSet options = new MutableDataSet();
```

Comment:
Seems like the code below is repeated in quite a few places. Can consider extracting it out as a common utility function so  we avoid duplicating code.